Sources/ApplicativeRouter/SyntaxRouter.swift

```diff
@@ -0,0 +1,159 @@
+import Foundation
+import Prelude
+import Optics
+
+// MARK: - Syntax Router
+
+// TODO: should this be generic over any monoid `M` instead of using `RequestData` directly?
+// TODO: generic over an error semigroup too?
+public struct Router<A> {
+  let parse: (RequestData) -> (rest: RequestData, match: A)?
+  let print: (A) -> RequestData?
+  let template: (A) -> RequestData?
+
+  public func match(request: URLRequest) -> A? {
+    return (self <% _end).parse(route(from: request))?.match
+  }
+
+  public func match(url: URL) -> A? {
+    return match(request: URLRequest(url: url))
+  }
+
+  public func match(string: String) -> A? {
+    return URL(string: string).flatMap(match(url:))
+  }
+
+  public func request(for a: A) -> URLRequest? {
+    return self.print(a).flatMap(request(from:))
+  }
+
+  public func url(for a: A) -> URL? {
+    return self.print(a).flatMap(request(from:)).flatMap { $0.url }
+  }
+
+  public func templateRequest(for a: A) -> URLRequest? {
+    return self.template(a).flatMap(request(from:))
+  }
+
+  public func templateUrl(for a: A) -> URL? {
+    return self.templateRequest(for: a).flatMap { $0.url }
+  }
+}
+
+// Functor
+
+extension Router {
+  public func map<B>(_ f: PartialIso<A, B>) -> Router<B> {
+    return f <¢> self
+  }
+
+  public static func <¢> <B> (lhs: PartialIso<A, B>, rhs: Router) -> Router<B> {
+    return Router<B>(
+      parse: { route in
+        guard let (rest, match) = rhs.parse(route) else { return nil }
+        return lhs.image(match).map { (rest, $0) }
+      },
+      print: lhs.preimage >-> rhs.print,
+      template: lhs.preimage >-> rhs.template
+    )
+  }
+}
+
+// Apply
+
+extension Router {
+  /// TODO: use `Tuple`?
+  /// Processes with the left and right side routers, and if they succeed returns the pair of their results.
+  public static func <%> <B> (lhs: Router, rhs: Router<B>) -> Router<(A, B)> {
```

Comment:
this is the alternate applicative. I decided to use `<%>` to represent it because 1) it's different from `<*>` in that it has a different signature and is right associative instead of left, and 2) although `<*>` induces a `<%>` the converse is not true (and in fact this router is such a counterexample).